cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -545,6 +678,63 @@ func getAltNames(cfg *kubeadmapi.MasterConfiguration) (*certutil.AltNames, error
 			altNames.IPs = append(altNames.IPs, ip)
 		} else if len(validation.IsDNS1123Subdomain(altname)) == 0 {
 			altNames.DNSNames = append(altNames.DNSNames, altname)
+		} else {
+			fmt.Printf("[certificates] WARNING: '%s' was not added to the '%s' SAN, because it is not a valid IP or RFC-1123 compliant DNS entry\n", altname, kubeadmconstants.APIServerCertName)
+		}
+	}
+
+	return altNames, nil
+}
+
+// getEtcdAltNames builds an AltNames object for generating the etcd server certificate
+// localhost is included in the SAN since this is the interface the etcd static pod listens on
+// hostname and API.AdvertiseAddress are excluded since etcd does not listen on this interface by default
+// the user can override the listen address with Etcd.ExtraArgs and add SANs with EtcdCertSANs
+func getEtcdAltNames(cfg *kubeadmapi.MasterConfiguration) (*certutil.AltNames, error) {
```

Comment:
this logic _could_ make sense in an other package, like pkiutil or the general util instead of here in a phase package.

Reply:
I like this suggestion.
It touches code that's outside of the scope of making this patch function.
Let's do a follow-up PR

Reply:
@luxas -- I ended up implementing this while considering @jamiehannaford's comments.
See 1bed0a154c.

The one part I don't like about it is that I've added user-facing warnings to util package.
Lmk what you think?